### ISSUE TITLE
Possible false-positive IDE0004 redundant cast warning

### ISSUE THREAD

**Issue by Gnbrkm41** [dotnet/roslyn#35340]
**Version Used**: 
VS 2019 16.0.2
**Steps to Reproduce**:

```cs
        public Task Method(string name, int r, int g, int b)
        {
            return Method(name, (float)r, (float)g, (float)b);
        }

        public Task Method(string name, float r, float g, float b)
        {
            var colour = new Color(r, g, b);

            return DoSomething(name, colour);
        }

        public Task DoSomething(string name, Color color)
        {
            return Task.CompletedTask;
        }
``` 

**Expected Behavior**:
No suggestion should be made regarding the removal of casting, as it would lead to recursion.

**Actual Behavior**:
![image](https://user-images.githubusercontent.com/42944058/56912283-de683f00-6aa6-11e9-8380-bb47a5364f19.png)

Suggestion is made to remove the cast.

**Comment by CyrusNajmabadi** [comment]
This is by design.  Any individual cast can be removed without changing the overload.  Hence why all are gray (since you might want to choose any of them to remove).  However, if you remove two of them, the third will become required and will not be marked as removable anymore.  i.e.:

![image](https://user-images.githubusercontent.com/4564579/56920528-4bb0ab80-6a78-11e9-884c-c4a70e106608.png)


Note that 'fix all' does the right thing here as well.  it will remove the first two, but leave the last around since removing it would break semantics.

**Comment by Gnbrkm41** [comment]
Right, thanks for clarification!
